tests/Game.class.test.js

```diff
@@ -0,0 +1,965 @@
+      const game2048 = new Game();
+
+      expect(game2048.getStatus()).toBe('idle');
+    });
+
+    it('should be "playing" after start', () => {
+      const game2048 = new Game();
+
+      game2048.start();
+
+      expect(game2048.getStatus()).toBe('playing');
+    });
+
+    it('should be "playing" when board is full but moves are still possible', () => {
+      const game2048 = new Game([
+        [128, 128, 0, 8],
+        [16, 8, 16, 32],
+        [8, 16, 32, 64],
+        [16, 32, 0, 128],
+      ]);
+
+      game2048.start();
+
+      expect(game2048.getStatus()).toBe('playing');
+    });
+
+    it('should be a "win" when a 2048 tile is created', () => {
+      const game2048 = new Game([
+        [0, 0, 0, 0],
+        [0, 0, 1024, 1024],
+        [0, 0, 0, 0],
+        [0, 0, 0, 0],
+      ]);
+
+      game2048.start();
+      game2048.moveRight();
+
+      expect(game2048.getStatus()).toBe('win');
+    });
+
+    it('should be a "lose" when no more moves are possible', () => {
+      const game2048 = new Game([
+        [128, 128, 0, 8],
+        [16, 8, 16, 32],
+        [8, 16, 32, 64],
+        [16, 32, 0, 128],
+      ]);
+
+      game2048.start();
+      game2048.moveLeft();
+
+      expect(game2048.getStatus()).toBe('lose');
+    });
+  });
+
+  describe('score', () => {
+    it('should be 0 by default', () => {
+      const game2048 = new Game();
+
+      expect(game2048.getScore()).toBe(0);
+    });
+
+    it('should be 0 for a custom state', () => {
+      const game2048 = new Game([
+        [16, 0, 32, 0],
+        [32, 0, 8, 8],
+        [0, 8, 8, 0],
+        [8, 8, 0, 16],
+      ]);
+
+      expect(game2048.getScore()).toBe(0);
+    });
+
+    it('should be 0 after start with a custom state', () => {
+      const game2048 = new Game([
+        [16, 0, 32, 0],
+        [32, 0, 8, 8],
+        [0, 8, 8, 0],
+        [8, 8, 0, 16],
+      ]);
+
+      game2048.start();
+
+      expect(game2048.getScore()).toBe(0);
+    });
+
+    it('should be updated after cells are merged horizontally', () => {
+      const game2048 = new Game([
+        [16, 0, 32, 0],
+        [32, 0, 8, 8],
+        [0, 8, 8, 0],
+        [8, 8, 0, 16],
+      ]);
+
+      game2048.start();
+      game2048.moveLeft();
+
+      expect(game2048.getScore()).toBe(48);
+    });
+
+    it('should be updated after cells are merged vertically', () => {
+      const game2048 = new Game([
+        [16, 0, 32, 0],
+        [32, 8, 8, 8],
+        [0, 8, 8, 0],
+        [8, 0, 0, 16],
+      ]);
+
+      game2048.start();
+      game2048.moveUp();
+
+      expect(game2048.getScore()).toBe(32);
+    });
+
+    it('should be updated after cells are merged multiple times per line', () => {
+      const game2048 = new Game([
+        [16, 16, 32, 32],
+        [32, 32, 8, 8],
+        [0, 8, 8, 8],
+        [8, 0, 16, 8],
+      ]);
+
+      game2048.start();
+      game2048.moveRight();
+
+      expect(game2048.getScore()).toBe(192);
+    });
```

Comment:
The test cases for the score should verify that the score is not only updated after merges but also that it accumulates correctly over multiple moves.